packages/components/src/utils/createComponent.js

```diff
@@ -1,21 +1,32 @@
 import { contextConnect } from '@wp-g2/context';
+import _ from 'lodash';
 import React from 'react';
 
 import { View } from '../View';
 
-export function createComponent({
-	as = 'div',
+/**
+ * @template {import('reakit-utils/types').As} T
+ * @template {import('@wp-g2/create-styles').ViewOwnProps<{}, T>} P
+ * @param {import('./types').Options<T, P>} options
+ * @return {import('@wp-g2/create-styles').PolymorphicComponent<T, import('./types').PropsFromViewOwnProps<P>>}
+ */
+export const createComponent = ({
+	as,
 	name = 'Component',
-	useHook = () => ({}),
+	useHook = _.identity,
 	memo = true,
-}) {
+}) => {
+	/**
+	 * @param {P} props
+	 * @param {import('react').Ref<T>} forwardedRef
+	 */
 	function Component(props, forwardedRef) {
 		const otherProps = useHook(props);
 
-		return <View as={as} {...otherProps} ref={forwardedRef} />;
+		return <View as={as || 'div'} {...otherProps} ref={forwardedRef} />;
```

Comment:
Neat. I'm glad this is done at the `createComponent` level, vs. having to do it for every component.